service/provider/src/main/java/gov/va/vro/service/provider/MasPollingProcessor.java

```diff
@@ -1,42 +1,156 @@
+        claimPayload.getCollectionId());
+    AbdEvidence abdEvidence = new AbdEvidence();
+    try {
+      var response =
+          masCollectionAnnotsApiService.getCollectionAnnots(claimPayload.getCollectionId());
+      for (MasCollectionAnnotation masCollectionAnnotation : response) {
+        log.info(
+            "Collection Annotation Response : Collection ID  {}",
+            masCollectionAnnotation.getCollectionsId());
+        log.info(
+            "Collection Status Response : Veteran FileId  {}  ",
+            masCollectionAnnotation.getVtrnFileId());
+
+        MasCollectionAnnotsResults masCollectionAnnotsResults = new MasCollectionAnnotsResults();
+        abdEvidence = masCollectionAnnotsResults.mapAnnotsToEvidence(masCollectionAnnotation);
+
+        log.info("AbdEvidence : Medications {}  ", abdEvidence.getMedications().size());
+        log.info("AbdEvidence : Conditions {}  ", abdEvidence.getConditions().size());
+        log.info("AbdEvidence : BP {}  ", abdEvidence.getBloodPressures().size());
+        break;
+      }
+    } catch (Exception e) {
+      log.error("Error in calling collection Status API ", e);
+      throw new MasException(e.getMessage(), e);
+    }
+    return abdEvidence;
+  }
+
+  public GeneratePdfResp generatePdf(MasAutomatedClaimPayload claimPayload, AbdEvidence abdEvidence)
+      throws MasException {
+
+    GeneratePdfPayload generatePdfPayload = new GeneratePdfPayload();
+    generatePdfPayload.setEvidence(abdEvidence);
+    generatePdfPayload.setClaimSubmissionId(claimPayload.getClaimDetail().getBenefitClaimId());
+    generatePdfPayload.setDiagnosticCode(
+        claimPayload.getClaimDetail().getConditions().getDiagnosticCode());
+    VeteranInfo veteranInfo = new VeteranInfo();
+    veteranInfo.setFirst(claimPayload.getFirstName());
+    veteranInfo.setLast(claimPayload.getLastName());
+    veteranInfo.setMiddle("");
+    veteranInfo.setBirthdate(claimPayload.getDateOfBirth());
+    generatePdfPayload.setVeteranInfo(veteranInfo);
+    log.info(
+        "Generating pdf for claim: {} and diagnostic code {}",
+        generatePdfPayload.getClaimSubmissionId(),
+        generatePdfPayload.getDiagnosticCode());
+    try {
+      log.info(generatePdfPayload.toString());
+      String response = camelEntrance.generatePdf(generatePdfPayload);
+      log.info(response.toString());
+      GeneratePdfResp pdfResponse = objectMapper.readValue(response, GeneratePdfResp.class);
+      log.info(pdfResponse.toString());
+      log.info("RESPONSE from generatePdf returned status: {}", pdfResponse.getStatus());
+      if (pdfResponse.getStatus().equals("ERROR")) {
+        log.info("RESPONSE from generatePdf returned error reason: {}", pdfResponse.getReason());
+        String errMsg =
+            "RESPONSE from generatePdf returned error reason: " + pdfResponse.getReason();
+        throw new MasException(errMsg, new MasException());
```

Comment:
See the comment above. We should be calling the pdf generation on the route not here. I hope that is the long term plan. I guess it is ok for now but doing the calls here like this is completely contrary to my understanding of how we want to use camel.